### DISCUSSION TITLE
Requesting clarifications in CLA

### DISCUSSION THREAD

**Discussion by n0toose** [audacity/audacity#2695] [Q&A]
Hi. I'll start off by fully clarifying my intentions -- I am interested in sending in a few patches that are either experimental UI/UX features that I published or have worked on in private, maybe a few patches that should make it more portable, as well as patches that should hypothetically improve the possibilities of Audacity being further developed under a derivative work, (if we were to assume that the current primary sponsors of it are to become suddenly unable of supporting it any further). My interests align solely with the development of software that fall under the category of public goods, while making sure that I have established a set of boundaries that would bar anyone from taking my work and using it for purposes that contradict these said goals or without me having any control over it.

I am particularly skeptical about signing the CLA, because I cannot understand if it is only effective when I submit contributions using either the "Pull requests" feature on GitHub or through some mailing list:
1. If I were to create a fork without submitting the changes that I want to submit as a pull request, does that mean that you guys would be able to incorporate those other changes under the same terms without me being able to do anything about it? (`and distribute your Contribution and such derivative works.`)
2. If I were to create any other project, regardless of whether or not it is relevant to audio processing and editing, would the `"perpetual, non-exclusive, worldwide, fully paid-up, royalty free, irrevocable copyright license to reproduce, prepare derivative works of, publicly display, publicly perform, sublicense, and distribute"` the hypothetical project still apply?
3. Can you define the term "sublicense" more clearly?

I believe that the answers to these questions are very obvious, however, it would seriously give me a peace of mind if these answers would be explicitly clarified in a legally binding document with consequences that are permanent. I think that it's worth reiterating again that I do not aim to set off a new flame war or controversy (I think that it comes without saying that you should lock this thread if something like that happens).

**Comment by crsib** [comment]
We will not take any changes unless their author has signed CLA.

>while making sure that I have established a set of boundaries that would bar anyone from taking my work and using it for purposes that contradict these said goals or without me having any control over it.

I'm not sure if this is possible under the GPL license. You probably need a different one, however you won't be able to relicense Audacity from the GPL2-or-later.

**Reply by n0toose** [reply]
> I'm not sure if this is possible under the GPL license. You probably need a different one, however you won't be able to relicense Audacity from the GPL2-or-later.

Well, I meant to the extent that the GPL allows. I'm willfully giving up some sort of control as soon as I use the GPL, as part of an agreement with obligations from the other party that are clear (e.g. "You will supply to me the code if I request it, even if it comes at a small cost").

> We will not take any changes unless their author has signed CLA.

I respect that there's a certain way that you work around here and that if I disagree with it, I am more than free to fork the software that you all develop and maintain that entire beast on my own (lol).

I am just particularly wary because I am not sure I either do not fully understand the terms of the CLA, simply because of a simple detail that I find vague.

I would sign a document that transfers a license to the company backing Audacity, but I am just, in all honesty and good faith, not sure where the line stops. Since the thing I wrote earlier probably did not make my question immediately clear, I'll try again with a hypothetical example: If I come up with my own set of Audacity plugins/effects and publish source code that I hold all the rights to ("proprietary source-available"), which are completely irrelevant to a set of patches that I have previously submitted to Audacity, will that count as a "Contribution", which means that by signing the CLA, I am also giving effectively handing over the rights to this hypothetical set of effects/add-ons? This may not necessarily be a deal breaker for me, it's whatever, but I am not sure what my end of the bargain here is -- simply because of what I perceive as a minor ambiguity with big consequences (everything else is clear).

**Reply by n0toose** [reply]
I would very much like to help with a thing or two, but if I cannot sure about what your document has me agree to as part of an extra set of conditions for something that's already, by definition, free work, then I am really not sure if my own altruism and enthusiasm can be enough of a factor for me to proceed with anything.

**Reply by nmggithub** [reply]
> If I come up with my own set of Audacity plugins/effects and publish source code that I hold all the rights to ("proprietary source-available"), which are completely irrelevant to a set of patches that I have previously submitted to Audacity, will that count as a "Contribution", which means that by signing the CLA, I am also giving effectively handing over the rights to this hypothetical set of effects/add-ons?

Not a legal expert, nor am I a member of the Audacity team. However, this seems to hinge on what counts as a "Contribution" under the CLA. Common sense would seem to indicate that the term only covers contributions explicitly given as such (in this case, submitting them as a PR).

Making patches and submitting them as a PR would fall under the CLA.

Making some plugins later that you publish on your own, even though they are related to Audacity, aren't explicitly given as Contributions (as they are not included in a PR). Thus, you would retain rights.

Again, not a member of the Audacity Team (and I hope I understand your question correctly), but that's my two cents.

To bring @crsib back in here (and to reword the question):

***If a Contributor signs the CLA, would any code they ever put on a public platform that is related to Audacity be considered a Contribution, or only those submitted through the PR process?***

**Reply by n0toose** [reply]
> Common sense would seem to indicate that the term only covers contributions explicitly given as such (in this case, submitting them as a PR).

You correctly understood what I meant and shorten it down to two sentences nicely, thanks. I completely agree with you on how "this is obvious" and share the same exact perspective on you, but what I am trying to say is that ambiguities "common sense" (assuming that the Audacity team also has the same exact interpretation of the document as the two of us do) in the context of a legally binding contract with an indefinite duration are absolutely unacceptable. I am not a lawyer, but come on, that's Contracts 101 and the work we're talking about here is literally free labor. All of these hypothetical scenarios (with various degrees of extremeness) were meant to illustrate scenarios where such ambiguities would appear so that .

What I seriously don't understand is, again, assuming that we all understand what is being written on the CLA the same way and presuming that the consequences (to the disadvantage of the prospective contributor), is why I would end up getting ghosted on over this. Jeez.

**Comment by SteveDaulton** [comment]
Disclosure / disclaimer: I'm not a lawyer, and I'm no longer a member of the Audacity team.This is just my 2c.

> 1. If I were to create a fork without submitting the changes that I want to submit as a pull request, does that mean that you guys would be able to incorporate those other changes under the same terms without me being able to do anything about it?

Legally, that depends entirely on the license terms that you release your code under. If your chosen license permits it, then yes it's permitted, if not it's not. However, Audacity will not use your code without the CLA, even if legally allowed (by your license) to do so.

A bit of background:
The Audacity team want to add support for VST3. That's not possible under the terms of GPL v2, but is allowed under GPL v3, so Audacity's license will be upgraded from GPL v2+ to GPL v3. The CLA allows Audacity to do that.




> 2. If I were to create any other project, regardless of whether or not it is relevant to audio processing and editing, would the `"perpetual, non-exclusive, worldwide, fully paid-up, royalty free, irrevocable copyright license to reproduce, prepare derivative works of, publicly display, publicly perform, sublicense, and distribute"` the hypothetical project still apply?

In short, the CLA ensures that if you give permission for Audacity to use your code (which is your choice), then you can't turn around at a later date and revoke that permission. The term "non-exclusive" is also important as it allows you to still do whatever you want with your code under any license terms that you want.

**Reply by Paul-Licameli** [reply]
Not a lawyer either but as I understand:  The CLA governs contributions that are merged into Audacity's repository.  It does not apply to what you do in any other project.
